sql/2022/third-parties/compressed_images_by_3p.sql

```diff
@@ -0,0 +1,73 @@
+#standardSQL
+# Compressed images by third parties
+
+WITH requests AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    pageid AS page,
+    url,
+    resp_content_encoding AS content_encoding,
+    type
+  FROM
+    `httparchive.summary_requests.2022_06_01_*`
+  WHERE
+    type = 'image' AND (
+      resp_content_encoding = 'gzip' OR
+      resp_content_encoding = 'br'
+    )
+),
+
+third_party AS (
+  SELECT
+    NET.HOST(domain) AS domain,
+    COUNT(DISTINCT page) AS page_usage
+  FROM
+    `httparchive.almanac.third_parties` tp
+  JOIN
+    requests r
+  ON NET.HOST(r.url) = NET.HOST(tp.domain)
+  WHERE
+    date = '2022-06-01' AND
+    category != 'hosting'
+  GROUP BY
+    domain
+  HAVING
+    page_usage >= 50
+)
+
+SELECT
+  client,
+  content_encoding,
+  domain,
+  num_requests,
+  total,
+  pct
+FROM (
+  SELECT
+    client,
+    content_encoding,
+    domain,
+    COUNT(0) AS num_requests,
+    SUM(COUNT(0)) OVER (PARTITION BY client, type) AS total,
+    COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY client, type) AS pct,
```

Comment:
What's this pct telling me? That Facebook serves 5.9% of all images used third parties used by all home pages that use br? I'm not sure how useful that is really? Or did you mean it as a percentage of each third party's own images?

Also why do we have any br images? Shouldn't the image format include compression and it's normally a bad practice to further compress. Except for SVGs I guess. In which case I further question the usefulness of this stat as some third-parties will use SVGs and some  not.

Confused.

Reply:
The 5% `pct` on Facebook refers to 5% of all compressed images are Facebook's. While it may be more appropriate (if possible) to show that 5% of all of Facebook's images are compressed.

I have also updated the query to exclude SVGs.

I have gone through the results and they seem correct, (e.g. https://www.s-coop-sumai.net/wp-content/uploads/sites/288/2021/01/fearure_3.png which is 58.5KB gzip compressed and 57.9KB uncompressed). Having said that, the increase in KBs is small and most of the images are tiny, with many 1x1.

Reply:
Maybe should limit it to images bigger than  X (for some value of X)? To ignore the 1x1 pixels where it probably doesn’t make a difference (some of these may even be empty IIRC?).

Reply:
I have added the image size as a column in the results and filter to the largest 100 third-parties for each content-encoding.